lib/rules/no-redundant-landmark-role.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Use of redundant or invalid role on <${element}> detected. If a landmark element is used, any role provided will either be redundant or incorrect.`;
+}
+
+const LANDMARK_ROLES = [
+  {
+    name: 'header',
+    role: 'banner',
+  },
+  {
+    name: 'main',
+    role: 'main',
+  },
+  {
+    name: 'aside',
+    role: 'complementary',
+  },
+  {
```

Comment:
@rajasegar also we'll want to disallow `<form role="form">`.

Reply:
done